---
title: "Ray_review"
output: html_document
---
## Code Proficiency 
  * Provided code from forked/cloned repository did not run sucessfully
  * However this may due to several reasons that do not reflect the profiency of the code, such as working directory issues and the environment the code was run in
  

## Input Test Page and Prep Phase
  * The use of the args function as way of quality control was great to see
  * The args function were passed to the Prep Phase

## Unlooped SNP List Work:
  * Using the 'unique' function to generate your chromosomal classification was efficent. 
  * The two list of either "present" or "absent" generated by way of the %in% operator to identify if an element belongs to a give vector was really intuitive
  * Would like to see how the syntax pulls out the two differnt groups between "maize" and "teosnite"

## Outer Loop
  * Very well explained and the purpose of the loop was detailed and explained concisely
  * Setting both the column names and the row names within this loop also was efficent
  * Using filter followed by transpose also was easily performed by this for loop. 
  
## Inner Loop 1
  `* Generation of the excluded false chromosome catergories was clearly detailed
  
## Inner Loop 2
  * Passing the args function previously defined

## Summary Review

This was efficiently written code using both the args function to pass the imported fang_et_al_genotype data along with a complex and concise for loop with two inner loops. I would have liked more explanation for my own clarity on how the syntax of how "maize" and "teosinte" data was parsed out. Also I am unclear on how the snp_position data was combined with the "maize" and "teosinte" data. For future assignments I will plan to add more detail to what the function and purpose of each line or block of code so that anyone not familiar with the syntax can more easily follow along.


The 40 output files were included in the repository however the repository did not address Part II of the assignment.

Finally, oing through the bulk of the code and trying to understand the syntax it seemed that perhaps steps were missed either with parsing the "maize" and "teosinte" groups out and then joining them downstream with the snp_position data along with graphical output.